src/services/BiometricAuth.js

```diff
@@ -0,0 +1,157 @@
+/**
+ * @author Arthur Martello <arthur.martello@etu.utc.fr>
+ *
+ * @copyright Copyright (c) 2019, SiMDE-UTC
+ * @license GPL-3.0
+ */
+
+import React from 'react';
+import { Modal, Platform, Text, View } from 'react-native';
+import * as LocalAuthentication from 'expo-local-authentication';
+import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome';
+import * as Haptics from 'expo-haptics';
+import { BlurView } from 'expo-blur';
+import { _, BiometricAuth as t } from '../utils/i18n';
+import colors from '../styles/colors';
+import BlockTemplate from '../components/BlockTemplate';
+
+export const defaultSecurity = ['transfer', 'refill', 'badge-locking'];
+export const advancedSecurity = ['transfer', 'refill', 'badge-locking', 'app-opening'];
+
+export default class BiometricAuth extends React.PureComponent {
+	static async hasHardware() {
+		return LocalAuthentication.hasHardwareAsync();
+	}
+
+	constructor(props) {
+		super(props);
+		this.state = {
+			showModal: false,
+			accepted: false,
+		};
+	}
+
+	async authenticate(successCallback, errorCallback, message = t('default_message')) {
```

Comment:
Hmmmmmmmmmmmm here, i am not sure about that, but I think that JS resolves `t('default_message')` at the parsing process, so if we change the language of the app, it will still with the previous loaded language :/